docs/04.guides/04.cookbooks/29.Threads-Scope/page.md

```diff
@@ -0,0 +1,272 @@
+		<cfset thread.start=now()>
+		<cfset sleep(1000+(index*10))>
+		<cfset thread.end=now()>
+	</cfthread>	
+</cffunction>
+
+<cfset start=getTickCount()>
+
+<cfloop from="1" to="5" index="index">
+	<cfset napASecond(index)>
+</cfloop>
+
+<!-- show threads -->
+<cfdump var="#cfthread#" expand="false">
+
+<!-- wait for all threads to finish -->
+<cfthread action="join" name="t1,t2,t3">
+
+<!-- show threads -->
+<cfdump var="#cfthread#" expand="false">
+
+<cfdump var="done in #getTickCount()-start#ms">
+
+```
+
+Here it join three threads "t1,t2,t3". We can join threads by its name.
+
+
+### Example 3 ###
+
+Here we can see about threads running parallel using ```each``` function.
+
+```lucee
+<cfscript>
+tasks = ["Susi","Sorglos"];
+start=getTickCount();
+tasks.each(
+	function(value, index, array){
+		systemOutput(value,true);
+		sleep(1000);
+	}
+	,true
+);
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+
+```each``` function has a option parallel, when we set parallel attribute as true. It runs each element parallel. In default it should be false.
+
+So that two array elements runs parallel done in 1second.
+
+
+```lucee
+<cfscript>
+tasks = [];
+loop from=1 to=100 index="i" {
+	arrayAppend(tasks,"t"&i)
+}
+start=getTickCount();
+tasks.each(
+	function(value, index, array){
+		systemOutput(value,true);
+		sleep(1000);
+	}
+	,true
+	//,100
+);
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+
+Here we have 100 array elements, setting option parallel as true it execute for 5 seconds not at 1 second because in default lucee runs 20 threads at the time after that it wait for free threads so only it take 5 seconds to complete.
+
+But we have option to set maximum threads open at that time. If it set to 100, it can open 100 threads at that time run parallel it take 1 second to complete.
+
+
+```each``` function is used not only for array, it used for struct & query also. Below example explain all the types.
+
+
+```lucee
+<cfscript>
+// ARRAY
+tasks = ["a","b"];
+tasks.each(
+	function(value, key, struct){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+arrayEach(
+	tasks
+	,function(value, key, struct){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+// STRUCT
+tasks = {a:1,b:2};
+structEach(
+	tasks
+	,function(value, key, struct){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+tasks.each(
+	function(value, key, struct){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+// QUERY
+persons = query(
+	'firstName':['Susi','Harry']
+	,'lastName':['Sorglos','Hirsch']
+);
+queryEach(
+	persons
+	,function(value, row, query){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+persons.each(
+	function(value, row, query){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+dump("done");
+</cfscript>
+```
+
+### Example 4 ###
+
+We are discussing for a long time how to extend functionality to make it easier to use or other new functionality.
+
+Here we explained about the future implementation of thread in lucee
```

Comment:
***ToChange***
***This is for lines 218 through 220***
Lucee members often discuss how to extend functionality to make Lucee easier to use or adding other new functionality.

This example shows a future implementation of threads in Lucee.